debian/stretch/foreman/foreman.dynflowd.init

```diff
@@ -0,0 +1,265 @@
+    if [  0$SECONDVAL -eq 0  -a  0$RETVAL -gt 0 ]; then
+       # some is running, some not
+       return 10
+    else
+       return $RETVAL
+    fi
+}
+
+status_q() {
+    kstatus &> /dev/null
+    return $?
+}
+
+start() {
+
+    echo -n $"Starting $NAME: "
+    status_q
+    STATUS=$?
+    if [ 0$STATUS -eq 0 ]; then
+        echo -n $"$NAME is already running. "
+        RETVAL=0
+        echo_success
+    elif [ 0$STATUS -eq 10 ]; then
+        echo -n $"Some processes of $NAME is already running. Some not. Please stop $NAME first."
+        RETVAL=10
+        log_end_msg 1
+    else
+        # rails expects you to run from the root of the app
+        pushd ${FOREMAN_HOME} >/dev/null
+        # start jobs
+        su -s /bin/bash - ${FOREMAN_USER} -c "\
+            RAILS_RELATIVE_URL_ROOT='$RAILS_RELATIVE_URL_ROOT' \
+            RAILS_ENV='$RAILS_ENV' \
+            FOREMAN_LOGGING='$FOREMAN_LOGGING' \
+            FOREMAN_LOGGING_SQL='$FOREMAN_LOGGING_SQL' \
+            $DAEMON $FOREMAN_TASK_PARAMS -m -n $FOREMAN_TASK_WORKERS start &>>${FOREMAN_LOG_DIR}/jobs-startup.log"
+        RETVAL=$?
+        if [ $RETVAL = 0 ]; then
+            log_success_msg $NAME start
+        else
+            log_failure_msg $NAME start
+        fi
+        popd >/dev/null
+    fi
+
+    echo
+    return $RETVAL
+}
+
+stop() {
+
+    status_q
+    STATUS=$?
+    if [ 0$STATUS -eq 0 -o 0$STATUS -eq 10 ]; then
+        # rails expects you to run from the root of the app
+        pushd ${FOREMAN_HOME} >/dev/null
+        su -s /bin/bash - ${FOREMAN_USER} -c "\
+            RAILS_RELATIVE_URL_ROOT='$RAILS_RELATIVE_URL_ROOT' \
+            RAILS_ENV='$RAILS_ENV' \
+            FOREMAN_LOGGING='$FOREMAN_LOGGING' \
+            FOREMAN_LOGGING_SQL='$FOREMAN_LOGGING_SQL' \
+            $DAEMON $FOREMAN_TASK_PARAMS -m -n $FOREMAN_TASK_WORKERS stop &>>${FOREMAN_LOG_DIR}/jobs-startup.log"
+        if [ -f $FOREMAN_PID_DIR/dynflow_executor_monitor.pid ]; then
+            echo -n $"Stopping dynflow_executor_monitor: "
+            killproc -p $FOREMAN_PID_DIR/dynflow_executor_monitor.pid
+            echo
+        fi
+        if [ -f $FOREMAN_PID_DIR/dynflow_executor.pid ]; then
+            echo -n $"Stopping dynflow_executor: "
+            killproc -p $FOREMAN_PID_DIR/dynflow_executor.pid
+            echo
+        fi
+       popd >/dev/null
+        RETVAL=0
+    else
+        RETVAL=0
+        echo -n "Stopping $NAME: "
+        echo_failure
+        echo
+    fi
+
+    return $RETVAL
+}
+
+restart() {
+    stop
+    start
+}
+
+condstop() {
+    status_q
+    STATUS=$?
+    if [ 0$STATUS -eq 0 -o 0$STATUS -eq 10 ]; then
+        stop
+    else
+        RETVAL=0
+    fi
+}
+
+condrestart() {
+    status_q
+    STATUS=$?
+    if [ 0$STATUS -eq 0 -o 0$STATUS -eq 10 ]; then
+        restart
+    else
+        RETVAL=0
+    fi
+}
+
+case "$1" in
+    start)
+        start
+        ;;
+    stop)
+        stop
+        ;;
+    restart)
```

Comment:
While probably not the topic of this PR, but Debian requires a `force-reload` target, that reloads the config *or* falls back to `restart`: https://www.debian.org/doc/debian-policy/#writing-the-scripts